MerlinAU.sh

```diff
@@ -6351,19 +6377,47 @@ _WAN_IsConnected_()
    return "$retCode"
 }
 
-##----------------------------------------##
-## Modified by Martinski W. [2024-Nov-13] ##
-##----------------------------------------##
+##------------------------------------------##
+## Modified by ExtremeFiretop [2024-Nov-15] ##
+##------------------------------------------##
 _PostUpdateEmailNotification_()
 {
    _DelPostUpdateEmailNotifyScriptHook_
+   FW_UpdateCheckState="$(nvram get firmware_check_enable)"
+   [ -z "$FW_UpdateCheckState" ] && FW_UpdateCheckState=0
+   if [ "$FW_UpdateCheckState" -eq 1 ] && [ "$sendEMailNotificationsFlag" ]
+   then
+       # Check if the CRON job already exists #
+       if ! eval $cronListCmd | grep -qE "$CRON_JOB_RUN #${CRON_JOB_TAG}#$"
+       then
+           if _AddCronJobEntry_
+           then
+               printf "Cron job '${GRNct}${CRON_JOB_TAG}${NOct}' was added successfully.\n"
+               _AddCronJobRunScriptHook_
+           else
+              printf "${REDct}**ERROR**${NOct}: Failed to add the cron job [${CRON_JOB_TAG}].\n"
+           fi
+       else
+           printf "Cron job '${GRNct}${CRON_JOB_TAG}${NOct}' already exists.\n"
+           _AddCronJobRunScriptHook_
+       fi
+   fi
    Update_Custom_Settings FW_New_Update_Changelog_Approval TBD
 
    local theWaitDelaySecs=10
-   local maxWaitDelaySecs=600  #10 minutes#
+   local maxWaitDelaySecs=720 #12 minutes#
```

Comment:
Increase to 12 minutes due to the how long it took my last test run:

![image](https://github.com/user-attachments/assets/87f9a0d2-ccb4-4b93-bad8-72e3c01bb5ac)

Reply:
> Increase to 12 minutes due to the how long it took my last test run:
> 
12 minutes!!! How exactly are you measuring the elapsed time? From which starting point?
I suspect there's something else going on. Will need more details.

Reply:
Well you can see that by the screenshot that it took a total of 14 minutes, and I assume 2 or 3 minutes of that was the install of the firmware and the reboot.

Checking the sent email on the gmail server shows the same time frame about a minute off from the email client above.

Keep in mind this is on a node, connected to another node via daisy chain, to the primary, so this is a node 2 nodes down from the primary:

![image](https://github.com/user-attachments/assets/e321192c-d2cc-4a8b-b17c-9a270d922052)

I'm assuming that either the additional WAN checks or the fact that it's a node is causing a slow down, and I don't want us to time out before it detects as "online".

Reply:
I also have my custom script on this node to restart the wireless service a few minutes after boot to fix the 6GHz radio. Maybe that has a part in it?

Reply:
> Well you can see that by the screenshot that it took a total of 14 minutes, and I assume 2 or 3 minutes of that was the install of the firmware and the reboot.
> 
> Checking the sent email on the gmail server shows the same time frame about a minute off from the email client above.
> 
>
What does the system log say about the timed entries from the "`_PostUpdateEmailNotification_`" function?
Those entries should give you the precise elapsed time from the code, without relying on other external events. Anything else would be just guessing.

Look for "**Post-Reboot Update Email Notification Wait Timeout**" in the system log.

Reply:
> I also have my custom script on this node to restart the wireless service a few minutes after boot to fix the 6GHz radio. Maybe that has a part in it?
>
>
Yeah, it's possible that by restarting the wireless service (from the "**services-start**" script ?) as part of the reboot sequence on your node, you're basically "pulling the rug from underneath" while the MerlinAU script is waiting for WAN connectivity, NTP synchronization & built-in services to get started.

To avoid such possible issues, make sure to put the call to restart the wireless service as the very **last** step in the reboot sequence with a sleep delay of 4-5 minutes to allow everyone else (i.e. all add-on script calls) to finish what they're supposed to do at startup **before** pulling the proverbial rug and cutting the connection.

My 2 cents.

Reply:
```
Nov 15 04:41:45 custom_script: Running /jffs/scripts/services-start
Nov 15 04:41:45 BHC: IN_BR eth0,R0 eth5,R0 eth8,R0 eth7,R0 eth6,R0
Nov 15 04:41:45 kernel: SCSI subsystem initialized
Nov 15 04:41:45 avahi-daemon[2544]: Service "GT-AXE11000" (/tmp/avahi/services/alexa.service) successfully established.
Nov 15 04:41:45 kernel: usb-storage 3-2:1.0: USB Mass Storage device detected
Nov 15 04:41:45 kernel: scsi host0: usb-storage 3-2:1.0
Nov 15 04:41:45 kernel: usbcore: registered new interface driver usb-storage
Nov 15 04:41:45 kernel: usbcore: registered new interface driver uas
Nov 15 04:41:45 kernel: tfat: loading out-of-tree module taints kernel.
Nov 15 04:41:45 kernel: Tuxera FAT 12/16/32 driver version 3017.7.24.32_1 [Flags: R/W MODULE].
Nov 15 04:41:45 kernel: Built against headers 4.1.52 #41 SMP PREEMPT Wed Apr 7 16:46:34 CST 2021 arm64
Nov 15 04:41:45 kernel: Running on kernel 4.1.52 #2 SMP PREEMPT Thu Nov 14 23:40:55 EST 2024 aarch64
Nov 15 04:41:45 kernel: Tuxera NTFS driver 3018.6.5.3_1 [Flags: R/W MODULE].
Nov 15 04:41:45 kernel: Built against headers 4.1.52 #41 SMP PREEMPT Wed Apr 7 16:46:34 CST 2021 arm64
Nov 15 04:41:45 kernel: Running on kernel 4.1.52 #2 SMP PREEMPT Thu Nov 14 23:40:55 EST 2024 aarch64
Nov 15 04:41:45 kernel: Tuxera HFS+ driver 3017.6.20
Nov 15 04:41:45 kernel: Built against headers 4.1.52 #41 SMP PREEMPT Wed Apr 7 16:46:34 CST 2021 arm64
Nov 15 04:41:45 kernel: Running on kernel 4.1.52 #2 SMP PREEMPT Thu Nov 14 23:40:55 EST 2024 aarch64
Nov 15 04:41:45 kernel: init (1): drop_caches: 1
Nov 15 04:41:45 : fwver: 3.0.0.4_388.8_3-g9b85de2f5e (sn:M9IG0X402953C2W /ha:04:42:1A:4F:4B:D0 )
Nov 15 04:41:45 [MerlinAU.sh] 2838: Post-update email notification hook was deleted successfully from '/jffs/scripts/services-start' script.
Nov 15 04:41:45 [MerlinAU.sh] 2838: Cron job hook was added successfully to '/jffs/scripts/services-start' script.
Nov 15 04:41:46 kernel: scsi 0:0:0:0: Direct-Access     Generic  USB Flash Disk   0.00 PQ: 0 ANSI: 6
Nov 15 04:41:46 kernel: sd 0:0:0:0: Attached scsi generic sg0 type 0
Nov 15 04:41:46 kernel: sd 0:0:0:0: [sda] 3934208 512-byte logical blocks: (2.01 GB/1.88 GiB)
Nov 15 04:41:46 kernel: sd 0:0:0:0: [sda] Write Protect is off
Nov 15 04:41:46 kernel: sd 0:0:0:0: [sda] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
Nov 15 04:41:46 kernel:  sda: sda1
Nov 15 04:41:46 kernel: sd 0:0:0:0: [sda] Attached SCSI removable disk
Nov 15 04:41:46 kernel: cfg80211: Calling CRDA to update world regulatory domain
Nov 15 04:41:47 BHC: Topology change from -1 to 1.
Nov 15 04:41:47 BHC: IN_BR 
Nov 15 04:41:49 kernel: device eth0 entered promiscuous mode
Nov 15 04:41:49 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:41:49 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:41:49 kernel: cfg80211: Calling CRDA to update world regulatory domain
Nov 15 04:41:51 BHC: IN_BR eth0,R2
Nov 15 04:41:52 rc_service: amas_lanctrl 2593:notify_rc restart_acsd
Nov 15 04:41:52 acsd: wl2.1: Selecting 6g band ACS policy
Nov 15 04:41:53 kernel: cfg80211: Calling CRDA to update world regulatory domain
Nov 15 04:41:53 custom_script: Running /jffs/scripts/pre-mount (args: /dev/sda1 ext4)
Nov 15 04:41:53 BONDING: option disabled
Nov 15 04:41:55 roamast: ROAMING Start...
Nov 15 04:41:56 kernel: cfg80211: Calling CRDA to update world regulatory domain
Nov 15 04:41:58 acsd: wl2.1: selected channel spec: 0x69cf (6g197/160)
Nov 15 04:41:58 acsd: wl2.1: Adjusted channel spec: 0x69cf (6g197/160)
Nov 15 04:41:58 acsd: wl2.1: selected channel spec: 0x69cf (6g197/160)
Nov 15 04:41:58 acsd: acs_set_chspec: 0x69cf (6g197/160) for reason APCS_INIT
Nov 15 04:41:59 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:42:00 httpd: Succeed to init SSL certificate...8443
Nov 15 04:42:02 rc_service: udhcpc_lan 4120:notify_rc stop_httpd
Nov 15 04:42:02 rc_service: udhcpc_lan 4120:notify_rc start_httpd
Nov 15 04:42:02 rc_service: waitting "stop_httpd" via udhcpc_lan ...
Nov 15 04:42:03 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5071 (6g113)
Nov 15 04:42:03 GT-AXE11000: start https:8443
Nov 15 04:42:03 GT-AXE11000: start httpd:80
Nov 15 04:42:03 lldpd[2101]: removal request for address of 192.168.50.1%26, but no knowledge of it
Nov 15 04:42:03 avahi-daemon[2544]: Withdrawing address record for 192.168.50.1 on br0.
Nov 15 04:42:03 avahi-daemon[2544]: Leaving mDNS multicast group on interface br0.IPv4 with address 192.168.50.1.
Nov 15 04:42:03 avahi-daemon[2544]: Interface br0.IPv4 no longer relevant for mDNS.
Nov 15 04:42:03 avahi-daemon[2544]: Joining mDNS multicast group on interface br0.IPv4 with address 192.168.50.3.
Nov 15 04:42:03 avahi-daemon[2544]: New relevant interface br0.IPv4 for mDNS.
Nov 15 04:42:03 avahi-daemon[2544]: Registering new address record for 192.168.50.3 on br0.IPv4.
Nov 15 04:42:03 rc_service: udhcpc_lan 4120:notify_rc start_dnsmasq
Nov 15 04:42:03 rc_service: waitting "start_httpd" via udhcpc_lan ...
Nov 15 04:42:03 httpd: Succeed to init SSL certificate...8443
Nov 15 04:42:05 rc_service: udhcpc_lan 4120:notify_rc stop_ntpd
Nov 15 04:42:05 rc_service: udhcpc_lan 4120:notify_rc start_ntpd
Nov 15 04:42:05 rc_service: waitting "stop_ntpd" via udhcpc_lan ...
Nov 15 04:42:06 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:42:06 ntpd: Started ntpd
Nov 15 04:42:07 LAN: network changes (192.168.50.3/255.255.255.0 --> 192.168.50.3/255.255.255.0)
Nov 15 04:42:07 rc_service: udhcpc_lan 4120:notify_rc stop_samba
Nov 15 04:42:07 Samba_Server: smb daemon is stopped
Nov 15 04:42:08 rc_service: udhcpc_lan 4120:notify_rc start_samba
Nov 15 04:42:08 Samba_Server: daemon is started
Nov 15 04:42:08 wsdd2[4723]: starting.
Nov 15 04:42:08 acsd: wl2.1: NONACSD channel switching to channel spec: 0x69cf (6g197/160)
Nov 15 04:42:47 ntpd: Initial clock set
Nov 15 04:42:47 rc_service: ntpd_synced 4731:notify_rc restart_diskmon
Nov 15 04:42:47 disk_monitor: Finish
Nov 15 04:42:47 disk_monitor: be idle
Nov 15 04:42:48 amtm disk check: Waited 16s for NTP to sync date
Nov 15 04:42:48 amtm disk check: Probing 'ext4' on device /dev/sda1
Nov 15 04:42:48 amtm disk check: Running disk check with command 'e2fsck -p' on /dev/sda1
Nov 15 04:42:48 amtm disk check: Disk check done on /dev/sda1
Nov 15 04:42:48 hotplug: USB ext4 fs at /dev/sda1 mounted on /tmp/mnt/USB3
Nov 15 04:42:48 kernel: EXT4-fs (sda1): mounted filesystem with ordered data mode. Opts: user_xattr
Nov 15 04:42:48 usb: USB ext4 fs at /dev/sda1 mounted on /tmp/mnt/USB3.
Nov 15 04:42:48 custom_script: Running /jffs/scripts/post-mount (args: /tmp/mnt/USB3)
Nov 15 04:42:48 Entware: Starting Entware services on /tmp/mnt/USB3
Nov 15 04:42:52 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5041 (6g65)
Nov 15 04:42:55 rc_service: cfg_client 5053:notify_rc restart_wireless;restart_qos;restart_firewall
Nov 15 04:42:56 kernel: device wl2.1 left promiscuous mode
Nov 15 04:42:56 kernel: br0: port 5(wl2.1) entered disabled state
Nov 15 04:42:56 get_ext_phy_id: 0/1/0/1
Nov 15 04:42:57 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:42:57 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:42:57 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:42:57 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:42:57 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:42:57 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:42:57 kernel: CFG80211-ERROR) wl_cfg80211_add_virtual_iface : Recvd application provided mac address:04:42:1a:4f:4b:d9
Nov 15 04:42:58 kernel: device wl2.1 entered promiscuous mode
Nov 15 04:42:58 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:42:58 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:42:58 lldpd[6198]: could not open either /etc/os-release or /usr/lib/os-release
Nov 15 04:42:58 lldpd[6198]: lsb_release information not available
Nov 15 04:42:58 lldpd[6202]: protocol LLDP enabled
Nov 15 04:42:58 lldpd[6202]: protocol CDPv1 disabled
Nov 15 04:42:58 lldpd[6202]: protocol CDPv2 disabled
Nov 15 04:42:58 lldpd[6202]: protocol SONMP disabled
Nov 15 04:42:58 lldpd[6202]: protocol EDP disabled
Nov 15 04:42:58 lldpd[6202]: protocol FDP disabled
Nov 15 04:42:58 lldpd[6202]: libevent 2.0.21-stable initialized with epoll method
Nov 15 04:42:58 lldpd[6202]: cannot get ethtool link information with GLINKSETTINGS (requires 4.9+): Operation not permitted
Nov 15 04:42:58 lldpcli[6199]: lldpd should resume operations
Nov 15 04:43:04 wpa_supplicant[6554]: Successfully initialized wpa_supplicant
Nov 15 04:43:04 wpa_supplicant[6554]: rfkill: Cannot open RFKILL control device
Nov 15 04:43:09 wlceventd: main(1239): wlceventd Start...
Nov 15 04:43:10 kernel: Cpuidle Host Clock divider is enabled
Nov 15 04:43:11 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:43:12 BHC: WiFi connection status change.
Nov 15 04:43:12 BHC: bandindex(0): state is 0
Nov 15 04:43:12 BHC: bandindex(1): state is 0
Nov 15 04:43:12 BHC: bandindex(2): state is 0
Nov 15 04:43:14 BHC: Topology change from -1 to 1.
Nov 15 04:43:15 kernel: device eth0 left promiscuous mode
Nov 15 04:43:15 kernel: br0: port 6(eth0) entered disabled state
Nov 15 04:43:16 BHC: IN_BR eth0,R0 eth5,R0 eth8,R0 eth7,R0 eth6,R0
Nov 15 04:43:19 roamast: ROAMING Start...
Nov 15 04:43:21 BHC: IN_BR 
Nov 15 04:43:26 kernel: device eth0 entered promiscuous mode
Nov 15 04:43:26 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:43:26 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:43:31 BHC: IN_BR eth0,R2
Nov 15 04:43:49 rc_service: amas_lanctrl 2593:notify_rc restart_acsd
Nov 15 04:43:49 acsd: wl2.1: Selecting 6g band ACS policy
Nov 15 04:43:50 acsd: acsd_main_loop(1174): sync_id: 14670, status:1, misc.error/abort
Nov 15 04:43:50 acsd: wl2.1: selected channel spec: 0x69af (6g165/160)
Nov 15 04:43:50 acsd: wl2.1: Adjusted channel spec: 0x69af (6g165/160)
Nov 15 04:43:50 acsd: wl2.1: selected channel spec: 0x69af (6g165/160)
Nov 15 04:43:50 acsd: acs_set_chspec: 0x69af (6g165/160) for reason APCS_INIT
Nov 15 04:43:50 acsd: acs_update_driver(451): acs update failed ret code: -16
Nov 15 04:43:50 acsd: acs_init_run(1100): wl2.1: update driver failed
Nov 15 04:43:55 acsd: wl2.1: NONACSD channel switching to channel spec: 0x50b9 (6g185)
Nov 15 04:44:00 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5001 (6g1)
Nov 15 04:44:01 rc_service: amas_lanctrl 2593:notify_rc stop_acsd
Nov 15 04:44:48 rc_service: hotplug 3432:notify_rc restart_nasapps
Nov 15 04:44:48 avahi-daemon[2544]: Files changed, reloading.
Nov 15 04:44:48 avahi-daemon[2544]: Loading service file /tmp/avahi/services/mt-daap.service.
Nov 15 04:44:48 avahi-daemon[2544]: Alias name GT-AXE11000: avahi_server_add_cname failure: The requested operation is invalid because redundant
Nov 15 04:44:48 iTunes: daemon is stopped
Nov 15 04:44:48 FTP_Server: daemon is stopped
Nov 15 04:44:48 wsdd2[4723]: terminating.
Nov 15 04:44:48 Samba_Server: smb daemon is stopped
Nov 15 04:44:48 avahi-daemon[2544]: Got SIGTERM, quitting.
Nov 15 04:44:48 avahi-daemon[2544]: Leaving mDNS multicast group on interface br0.IPv4 with address 192.168.50.3.
Nov 15 04:44:48 avahi-daemon[2544]: Leaving mDNS multicast group on interface lo.IPv4 with address 127.0.1.1.
Nov 15 04:44:48 avahi-daemon[2544]: avahi-daemon 0.8 exiting.
Nov 15 04:44:52 Timemachine: daemon is stopped
Nov 15 04:44:52 avahi-daemon[9504]: Found user 'nobody' (UID 65534) and group 'nobody' (GID 65534).
Nov 15 04:44:52 avahi-daemon[9504]: Successfully dropped root privileges.
Nov 15 04:44:52 avahi-daemon[9504]: avahi-daemon 0.8 starting up.
Nov 15 04:44:52 avahi-daemon[9504]: Loading service file /tmp/avahi/services/alexa.service.
Nov 15 04:44:52 avahi-daemon[9504]: Loading new alias name GT-AXE11000.
Nov 15 04:44:52 avahi-daemon[9504]: Joining mDNS multicast group on interface br0.IPv4 with address 192.168.50.3.
Nov 15 04:44:52 avahi-daemon[9504]: New relevant interface br0.IPv4 for mDNS.
Nov 15 04:44:52 avahi-daemon[9504]: Joining mDNS multicast group on interface lo.IPv4 with address 127.0.1.1.
Nov 15 04:44:52 avahi-daemon[9504]: New relevant interface lo.IPv4 for mDNS.
Nov 15 04:44:52 avahi-daemon[9504]: Network interface enumeration completed.
Nov 15 04:44:52 avahi-daemon[9504]: Registering new address record for 192.168.50.3 on br0.IPv4.
Nov 15 04:44:52 avahi-daemon[9504]: Registering new address record for 127.0.1.1 on lo.IPv4.
Nov 15 04:44:52 avahi-daemon[9504]: Registering new address record for 127.0.0.1 on lo.IPv4.
Nov 15 04:44:52 Samba_Server: daemon is started
Nov 15 04:44:52 wsdd2[9518]: starting.
Nov 15 04:44:53 avahi-daemon[9504]: Server startup complete. Host name is GT-AXE11000.local. Local service cookie is 2021708752.
Nov 15 04:44:53 avahi-daemon[9504]: Alias name GT-AXE11000: avahi_server_add_cname failure: The requested operation is invalid because redundant
Nov 15 04:44:54 avahi-daemon[9504]: Service "GT-AXE11000" (/tmp/avahi/services/alexa.service) successfully established.
Nov 15 04:46:45 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:46:47 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:46:47 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:47 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:48 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:48 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:49 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:49 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:50 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Receive scan result failed.
Nov 15 04:46:50 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Receive scan result failed.
Nov 15 04:46:50 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:50 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:50 AMAS_SSD: [do_site_survey] [pid:10689] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:46:51 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:51 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:51 AMAS_SSD: [escan_main_loop] [pid:10689] Scan failed beacuse scan timeout.
Nov 15 04:46:51 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] [eth7] unit(1) Set escan command failed.
Nov 15 04:46:51 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] [eth6] unit(0) Set escan command failed.
Nov 15 04:46:52 AMAS_SSD: [wl_get_scan_results_escan] [pid:10690] Stop scan for [eth7] unit(1)
Nov 15 04:46:52 AMAS_SSD: [wl_get_scan_results_escan] [pid:10688] Stop scan for [eth6] unit(0)
Nov 15 04:46:54 AMAS_SSD: [do_site_survey] [pid:10689] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:47:23 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:47:25 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:48:04 rc_service: service 11410:notify_rc restart_wireless
Nov 15 04:48:06 kernel: device wl2.1 left promiscuous mode
Nov 15 04:48:06 kernel: br0: port 5(wl2.1) entered disabled state
Nov 15 04:48:06 get_ext_phy_id: 0/1/0/1
Nov 15 04:48:07 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:07 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:07 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:07 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:07 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:07 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:07 kernel: CFG80211-ERROR) wl_cfg80211_add_virtual_iface : Recvd application provided mac address:04:42:1a:4f:4b:d9
Nov 15 04:48:08 kernel: device wl2.1 entered promiscuous mode
Nov 15 04:48:08 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:48:08 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:48:08 lldpd[12424]: could not open either /etc/os-release or /usr/lib/os-release
Nov 15 04:48:08 lldpd[12424]: lsb_release information not available
Nov 15 04:48:08 lldpd[12428]: protocol LLDP enabled
Nov 15 04:48:08 lldpd[12428]: protocol CDPv1 disabled
Nov 15 04:48:08 lldpd[12428]: protocol CDPv2 disabled
Nov 15 04:48:08 lldpd[12428]: protocol SONMP disabled
Nov 15 04:48:08 lldpd[12428]: protocol EDP disabled
Nov 15 04:48:08 lldpd[12428]: protocol FDP disabled
Nov 15 04:48:08 lldpd[12428]: libevent 2.0.21-stable initialized with epoll method
Nov 15 04:48:08 lldpd[12428]: cannot get ethtool link information with GLINKSETTINGS (requires 4.9+): Operation not permitted
Nov 15 04:48:08 lldpcli[12426]: lldpd should resume operations
Nov 15 04:48:15 wpa_supplicant[12939]: Successfully initialized wpa_supplicant
Nov 15 04:48:15 wpa_supplicant[12939]: rfkill: Cannot open RFKILL control device
Nov 15 04:48:19 kernel: CFG80211-ERROR) wl_cfg80211_bss_up : 'bss 1' failed with -1
Nov 15 04:48:19 kernel: CFG80211-ERROR) wl_cfg80211_bcn_bringup_ap : AP Bring up failed for bssidx 1
Nov 15 04:48:19 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:48:19 wlceventd: main(1239): wlceventd Start...
Nov 15 04:48:21 rc_service: amas_lanctrl 2593:notify_rc start_cfgsync
Nov 15 04:48:23 BHC: WiFi connection status change.
Nov 15 04:48:23 BHC: bandindex(0): state is 0
Nov 15 04:48:23 BHC: bandindex(1): state is 0
Nov 15 04:48:23 BHC: bandindex(2): state is 0
Nov 15 04:48:25 BHC: Topology change from -1 to 1.
Nov 15 04:48:26 kernel: device eth0 left promiscuous mode
Nov 15 04:48:26 kernel: br0: port 6(eth0) entered disabled state
Nov 15 04:48:26 BHC: IN_BR eth0,R0 eth5,R0 eth8,R0 eth7,R0 eth6,R0
Nov 15 04:48:29 roamast: ROAMING Start...
Nov 15 04:48:30 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:48:31 BHC: IN_BR 
Nov 15 04:48:36 kernel: device eth0 entered promiscuous mode
Nov 15 04:48:36 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:48:36 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:48:40 rc_service: cfg_client 14186:notify_rc restart_wireless;restart_qos;restart_firewall
Nov 15 04:48:42 kernel: device wl2.1 left promiscuous mode
Nov 15 04:48:42 kernel: br0: port 5(wl2.1) entered disabled state
Nov 15 04:48:42 get_ext_phy_id: 0/1/0/1
Nov 15 04:48:43 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:43 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:43 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:43 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:43 kernel: CFG80211-ERROR) dhd_config_dongle : Dongle is already up
Nov 15 04:48:43 kernel: CFG80211-ERROR) wl_allocate_wowl_config : Wowl not supported (-23)
Nov 15 04:48:43 kernel: CFG80211-ERROR) wl_cfg80211_add_virtual_iface : Recvd application provided mac address:04:42:1a:4f:4b:d9
Nov 15 04:48:44 kernel: device wl2.1 entered promiscuous mode
Nov 15 04:48:44 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:48:44 kernel: br0: port 5(wl2.1) entered forwarding state
Nov 15 04:48:44 lldpd[15996]: could not open either /etc/os-release or /usr/lib/os-release
Nov 15 04:48:44 lldpd[15996]: lsb_release information not available
Nov 15 04:48:44 lldpd[16000]: protocol LLDP enabled
Nov 15 04:48:44 lldpd[16000]: protocol CDPv1 disabled
Nov 15 04:48:44 lldpd[16000]: protocol CDPv2 disabled
Nov 15 04:48:44 lldpd[16000]: protocol SONMP disabled
Nov 15 04:48:44 lldpd[16000]: protocol EDP disabled
Nov 15 04:48:44 lldpd[16000]: protocol FDP disabled
Nov 15 04:48:44 lldpd[16000]: libevent 2.0.21-stable initialized with epoll method
Nov 15 04:48:44 lldpd[16000]: cannot get ethtool link information with GLINKSETTINGS (requires 4.9+): Operation not permitted
Nov 15 04:48:44 lldpcli[15998]: lldpd should resume operations
Nov 15 04:48:51 wpa_supplicant[16352]: Successfully initialized wpa_supplicant
Nov 15 04:48:51 wpa_supplicant[16352]: rfkill: Cannot open RFKILL control device
Nov 15 04:48:54 wlceventd: main(1239): wlceventd Start...
Nov 15 04:48:55 kernel: Cpuidle Host Clock divider is enabled
Nov 15 04:48:57 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:48:58 BHC: WiFi connection status change.
Nov 15 04:48:58 BHC: bandindex(0): state is 0
Nov 15 04:48:58 BHC: bandindex(1): state is 0
Nov 15 04:48:58 BHC: bandindex(2): state is 0
Nov 15 04:49:00 BHC: Topology change from -1 to 1.
Nov 15 04:49:02 kernel: device eth0 left promiscuous mode
Nov 15 04:49:02 kernel: br0: port 6(eth0) entered disabled state
Nov 15 04:49:02 BHC: IN_BR eth0,R0 eth5,R0 eth8,R0 eth7,R0 eth6,R0
Nov 15 04:49:04 roamast: ROAMING Start...
Nov 15 04:49:07 BHC: IN_BR 
Nov 15 04:49:12 kernel: device eth0 entered promiscuous mode
Nov 15 04:49:12 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:49:12 kernel: br0: port 6(eth0) entered forwarding state
Nov 15 04:49:17 BHC: IN_BR eth0,R2
Nov 15 04:49:35 rc_service: amas_lanctrl 2593:notify_rc restart_acsd
Nov 15 04:49:35 acsd: wl2.1: Selecting 6g band ACS policy
Nov 15 04:49:36 acsd: acsd_main_loop(1174): sync_id: 11189, status:1, misc.error/abort
Nov 15 04:49:36 acsd: wl2.1: selected channel spec: 0x69cf (6g197/160)
Nov 15 04:49:36 acsd: wl2.1: Adjusted channel spec: 0x69cf (6g197/160)
Nov 15 04:49:36 acsd: wl2.1: selected channel spec: 0x69cf (6g197/160)
Nov 15 04:49:36 acsd: acs_set_chspec: 0x69cf (6g197/160) for reason APCS_INIT
Nov 15 04:49:36 acsd: acs_update_driver(451): acs update failed ret code: -16
Nov 15 04:49:36 acsd: acs_init_run(1100): wl2.1: update driver failed
Nov 15 04:49:41 acsd: wl2.1: NONACSD channel switching to channel spec: 0x50b9 (6g185)
Nov 15 04:49:46 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5001 (6g1)
Nov 15 04:49:51 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5089 (6g137)
Nov 15 04:49:56 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5001 (6g1)
Nov 15 04:50:01 acsd: wl2.1: NONACSD channel switching to channel spec: 0x5059 (6g89)
Nov 15 04:50:05 rc_service: amas_lanctrl 2593:notify_rc stop_acsd
Nov 15 04:50:05 kernel: cfg80211: Exceeded CRDA call max attempts. Not calling CRDA
Nov 15 04:52:25 [MerlinAU.sh] 2838: Post-Reboot Update Email Notification Wait Timeout [600 sec.] expired.
Nov 15 04:52:31 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:52:33 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:52:33 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:33 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:34 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:34 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:35 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:35 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Receive scan result failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Receive scan result failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [do_site_survey] [pid:20903] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:52:37 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:37 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:38 AMAS_SSD: [escan_main_loop] [pid:20903] Scan failed beacuse scan timeout.
Nov 15 04:52:38 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:38 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:39 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] Stop scan for [eth7] unit(1)
Nov 15 04:52:39 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] Stop scan for [eth6] unit(0)
Nov 15 04:52:40 AMAS_SSD: [do_site_survey] [pid:20903] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:53:10 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:53:12 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:53:30 [MerlinAU.sh] 2838: The email notification was sent successfully [POST_REBOOT_FW_UPDATE_STATUS].
```

Reply:
> > I also have my custom script on this node to restart the wireless service a few minutes after boot to fix the 6GHz radio. Maybe that has a part in it?
> 
> Yeah, it's possible that by restarting the wireless service (from the "**services-start**" script ?) as part of the reboot sequence on your node, you're basically "pulling the rug from underneath" while the MerlinAU script is waiting for WAN connectivity, NTP synchronization & built-in services to get started.
> 
> To avoid such possible issues, make sure to put the call to restart the wireless service as the very **last** step in the reboot sequence with a sleep delay of 4-5 minutes to allow everyone else (i.e. all add-on script calls) to finish what they're supposed to do at startup **before** pulling the proverbial rug and cutting off the connection.
> 
> My 2 cents.

I was getting tired yesterday lol. My script in theory should have no impact because I am only restarting the wireless radios but this node is connected via the WAN interface (physically with Ethernet) so while I do have a script that restarts the wireless service, it shouldn't matter?

This is the logs of my script on that run:

```
2024-11-15 04:42:48 - Starting 6GHz Fix
2024-11-15 04:42:48 - SLEEPING
2024-11-15 04:47:48 - Current user: 
2024-11-15 04:47:48 - Current directory: /
2024-11-15 04:47:48 - Shell: 
2024-11-15 04:47:48 - Debug: Found wlc_closed_values: wlc2_closed=1
2024-11-15 04:47:48 - Debug: Processing var_name: wlc2_closed
2024-11-15 04:47:48 - Debug: Successfully set wlc2_closed to 0
2024-11-15 04:47:48 - Debug: Found wlc_ssid_values: wlc2_ssid=Whose Line Is It Anyways?-5GHz_dwb
2024-11-15 04:47:48 - Debug: Processing var_name: wlc2_ssid
2024-11-15 04:47:48 - Debug: Current value of wlc2_ssid: Whose Line Is It Anyways?-5GHz_dwb
2024-11-15 04:47:48 - Debug: Value after removing '_dwb': Whose Line Is It Anyways?-5GHz
2024-11-15 04:47:48 - Debug: Value after replacing with 6GHz: Whose Line Is It Anyways?-6GHz
2024-11-15 04:47:48 - Debug: wl_ssid_value updated to: Whose Line Is It Anyways?-6GHz
2024-11-15 04:47:48 - Debug: Set wl2.1_closed to 0
2024-11-15 04:47:48 - Debug: Set wl2.1_ssid to Whose Line Is It Anyways?-6GHz
2024-11-15 04:47:51 - Debug: NVRAM committed
2024-11-15 04:47:54 - Debug: Wireless interface wl2.1 ssid set to Whose Line Is It Anyways?-6GHz
2024-11-15 04:48:04 - Debug: Wireless service restarted
2024-11-15 04:48:14 - 6GHz Fix Done
```

We can cross reference that with the system logs above to see when it restarted the radio. Right now it's in post-mount actually, not in services-start. But I think the idea is valid so I'll move it there now :)

Reply:
OH! I FOUND THE PROBLEM!!!!
I never made my script a background job in post-mount, and it has 5 minute sleep.
So it causes everything after it to wait since it's not a background job, waiting 5 minutes for returned results.

Reply:
Now that we know the cause; we can do the math again to check.

14 minutes to send the email.
Assumption is 3 minutes of that is install and reboot

That leaves 11 minutes unaccounted for. We now know that 5 minutes of that 11 is my wait and restart of the wireless service. That means there's now 6 minutes unaccounted for. (Not flashing, rebooting, or waiting)

We can probably safely set this back to 8 or 10 minutes, but I'd keep it higher than 6 minutes.
Any thoughts on the fact that it sent the email after the timeout expired?

Reply:
Well; did it improve? Sure; but it's not the problem clearly....
I retested now with the latest version, and reflashed and instead of 14 minutes it's now 12 minutes.

![image](https://github.com/user-attachments/assets/52d4a3b2-5505-4756-b8fb-c4f1d5c649ae)

It's still waiting until it expires?

```
Nov 15 04:52:25 [MerlinAU.sh] 2838: Post-Reboot Update Email Notification Wait Timeout [600 sec.] expired.
Nov 15 04:52:31 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:52:33 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:52:33 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:33 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:34 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:34 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:35 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:35 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Receive scan result failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Receive scan result failed.
Nov 15 04:52:36 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:36 AMAS_SSD: [do_site_survey] [pid:20903] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:52:37 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:37 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:38 AMAS_SSD: [escan_main_loop] [pid:20903] Scan failed beacuse scan timeout.
Nov 15 04:52:38 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] [eth7] unit(1) Set escan command failed.
Nov 15 04:52:38 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] [eth6] unit(0) Set escan command failed.
Nov 15 04:52:39 AMAS_SSD: [wl_get_scan_results_escan] [pid:20905] Stop scan for [eth7] unit(1)
Nov 15 04:52:39 AMAS_SSD: [wl_get_scan_results_escan] [pid:20904] Stop scan for [eth6] unit(0)
Nov 15 04:52:40 AMAS_SSD: [do_site_survey] [pid:20903] unit(2) Scan failed, total scan results counts = 0 
Nov 15 04:53:10 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ sta_autoconnect 0
Nov 15 04:53:12 PtyConn: [SUCCESS] wpa_cli-2.7 -p /var/run/wl2_wpa_supplicant/ disconnect
Nov 15 04:53:30 [MerlinAU.sh] 2838: The email notification was sent successfully [POST_REBOOT_FW_UPDATE_STATUS].
```

I'll check the new nvram values now on the router. Maybe one doesn't update on the node?

Reply:
The problem is this function returning 1 currently on the node even though it has network access:

```
_WAN_IsConnected_()
{
   local retCode=1
   for iFaceNum in 0 1
   do
       if [ "$(nvram get wan${iFaceNum}_primary)" -eq 1 ] && \
          [ "$(nvram get wan${iFaceNum}_state_t)" -eq 2 ]
       then echo=0 ; break ; fi
   done
   return "$retCode"
}
```

![image](https://github.com/user-attachments/assets/08ef5152-e252-45c5-82f7-328e365ece4f)

Reply:
The problem is this value:

`wan0_state_t=0` is not flipping to "2" as expected in the script.
I am not sure if there is much value in this function now that we know the root cause of the problem was related to the lockfile being triggered on the second instance.

I'll remove the entire function and try again, on the node the ntp_ready value updates accordingly as we expect.

```
Admin@GT-AXE11000:/tmp/home/root# nvram get ntp_ready
1
```

Reply:
Bingo!!! Within 4 minutes now on the node:

![image](https://github.com/user-attachments/assets/a287c487-82f3-48f8-8e6c-694574645eb0)

Reply:
Dropped the wait back down to 8 minutes (double) what it should usually take.
Should be good enough for most connection types and routers.

Reply:
> Now that we know the cause; we can do the rough math again to check.
> 
> 14 minutes to send the email. Assumption is 3 minutes of that is install and reboot
> 
> That leaves 11 minutes unaccounted for. We now know that 5 to 6 minutes of that 11 is my wait and restart of the wireless service. That means there's now 6 to 5 minutes unaccounted for for that node. (Not flashing, rebooting, or waiting)
> 
> We can probably safely set this back to 8 or 10 minutes, but I'd keep it higher than 6 minutes. Any thoughts on the fact that it sent the email after the timeout expired?
>
>
Yeah, the email notification is sent even if the wait timeout has expired. It was coded that way just in case there was a chance that after the small sleep delay call just before it, the system was actually ready to send the email. I seems like in some instances that worked to our advantage, LOL :>).

Reply:
> > Now that we know the cause; we can do the rough math again to check.
> > 14 minutes to send the email. Assumption is 3 minutes of that is install and reboot
> > That leaves 11 minutes unaccounted for. We now know that 5 to 6 minutes of that 11 is my wait and restart of the wireless service. That means there's now 6 to 5 minutes unaccounted for for that node. (Not flashing, rebooting, or waiting)
> > We can probably safely set this back to 8 or 10 minutes, but I'd keep it higher than 6 minutes. Any thoughts on the fact that it sent the email after the timeout expired?
> 
> Yeah, the email notification is sent even if the wait timeout has expired. It was coded that way just in case there was a chance that after the small sleep delay call just before it, the system was actually ready to send the email. I seems like in some instances that worked to our advantage, LOL :>).

It sure did!

Because I sent in the end, it made it pretty clear one of the nvram values wasn't being updated on the nodes hehehe

Reply:
> The problem is this value:
> 
> `wan0_state_t=0`
> 
> is not flipping to "2" as expected in the script.
> 
>
Excellent catch!!!
I seem to recall that in the past, I read in some forum posts where some people have reported that when the router is operating as an AiMesh node, some NVRAM vars tend to have different meanings or different vars must be queried to get the same information. Unfortunately, I've never seen this documented anywhere by RMerlin, probably because the ASUS AiMesh module is "proprietary" so he has no access to its source code.

And that's probably one of the reasons that most add-on devs say their scripts are not "guaranteed" to work in AiMesh nodes - things seem to get a little screwy when getting some NVRAM values, and it happens **only** when AiMesh is involved.

>
> I am not sure if there is much value in this function now that we know the root cause of the problem was related to the lockfile being triggered on the second instance.
> 
> I'll remove the entire function and try again, on the node the ntp_ready value updates accordingly as we expect.
> 
> ```
> Admin@GT-AXE11000:/tmp/home/root# nvram get ntp_ready
> 1
> ```
Yes, that makes sense. The original set of conditions I had initially put in the functions is really all we need to determine that we're in a good state to proceed with the email notification. As explained before, if NTP is ready that means we have an active WAN connection and the system clock has been synced, so we're good to go once the built-in services are started.

Reply:
> Dropped the wait back down to 8 minutes (double) what it should usually take. Should be good enough for most connection types and routers.
>
>
In a couple of my personal custom scripts, I have a similar wait timeout loop where I wait until NTP is ready and built-in services have been restarted (in fact, I ported that code over to the MerlinAU script). Originally, the MAX timeout was 6 minutes, but on two rare occasions, the reboot sequence took longer - a little bit over 6 minutes - and I never figured out why, so I just decided to increase the MAX timeout to 10 minutes and I've never had any issues after that (and it has been about almost 4 years since then, running on several router models from friends & relatives).

My suggestion would be to leave the MAX to 10 minutes to account for those rare, unusual cases where, for some reason, the reboot sequence takes longer than expected. Perhaps it has something to do with the modem doing something else at the time and delaying the WAN connection and, therefore, delaying NTP syncing the clock. If that's the case, the extra cushion in the timeout can only help, IMO.

Reply:
@Martinski4GitHub 

No problemo, I trust your experiences the same way you trust mine!
Done in commit: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/355/commits/684665e68e8ad9f154c0e691905aa779e4e2caed

:D